### DISCUSSION TITLE
Bagaimana cara menampilkan informasi dari kolom tertentu pada tooltip plot_ly?

### DISCUSSION THREAD

**Discussion by Hertzspr** [teamalgoritma/community#91] [Q&A] [answered]
Saya punya sebuah data frame long form, dengan dput berikut: 
`structure(list(top_year = structure(c(1L, 1L, 1L, 1L, 1L, 1L, 
1L, 1L, 1L, 1L, 1L, 1L, 1L, 1L, 1L, 1L, 1L, 1L, 1L, 1L), .Label = c("2010", 
"2011", "2012", "2013", "2014", "2015", "2016", "2017", "2018", 
"2019"), class = "factor"), title = c("Talking to the Moon", 
"Talking to the Moon", "Talking to the Moon", "Talking to the Moon", 
"Talking to the Moon", "Talking to the Moon", "Talking to the Moon", 
"Talking to the Moon", "Talking to the Moon", "Talking to the Moon", 
"Love The Way You Lie", "Love The Way You Lie", "Love The Way You Lie", 
"Love The Way You Lie", "Love The Way You Lie", "Love The Way You Lie", 
"Love The Way You Lie", "Love The Way You Lie", "Love The Way You Lie", 
"Love The Way You Lie"), features = c("pop", "bpm", "nrgy", "dnce", 
"d_b", "live", "val", "dur", "acous", "spch", "pop", "bpm", "nrgy", 
"dnce", "d_b", "live", "val", "dur", "acous", "spch"), values = c(1.38058880374642, 
0.942830223863706, -0.65796172359903, -1.28609218834351, 0.327371183226631, 
-0.514536306631187, -1.9895286394006, -0.0602589528426108, 1.88295172112404, 
-0.761474145367176, 1.38058880374642, -1.30581490540942, 1.47217526005808, 
0.619116670899662, 0.327371183226631, 2.5379869999639, 0.607464526409839, 
1.06677881852791, 0.495064508917191, 1.3944549185263), description = c("Popularity scale of 1.38", 
"Tempo scale of 0.94", "Energetic scale of -0.66", "Danceablity scale of -1.29", 
"Relative loudness scale of 0.33", "Possibility it's a live recording scale of -0.51", 
"Positive mood scale of -1.99", "Duration scale of -0.06", "Accoustic tendency scale of 1.88", 
"Spoken word tendency scale of -0.76", "Popularity scale of 1.38", 
"Tempo scale of -1.31", "Energetic scale of 1.47", "Danceablity scale of 0.62", 
"Relative loudness scale of 0.33", "Possibility it's a live recording scale of 2.54", 
"Positive mood scale of 0.61", "Duration scale of 1.07", "Accoustic tendency scale of 0.5", 
"Spoken word tendency scale of 1.39")), class = c("grouped_df", 
"tbl_df", "tbl", "data.frame"), row.names = c(NA, -20L), groups = structure(list(
    top_year = structure(1L, .Label = c("2010", "2011", "2012", 
    "2013", "2014", "2015", "2016", "2017", "2018", "2019"), class = "factor"), 
    .rows = structure(list(1:20), ptype = integer(0), class = c("vctrs_list_of", 
    "vctrs_vctr", "list"))), class = c("tbl_df", "tbl", "data.frame"
), row.names = c(NA, -1L), .drop = TRUE))`

-

Saya berhasil membuat heatmap menggunakan plot_ly atas dataframe tersebut, dengan code berikut: 
`plot_ly(data=long_comp_byyear_bypop,
        z=~values, 
        x=~top_year, 
        y=~features, 
        type = "heatmap")`
![Untitled](https://user-images.githubusercontent.com/95233522/167693754-30891f29-d724-4ed8-8a17-da7d3d541c17.jpg)

Namun saya bingung bagaimana jika saya ingin menambahkan tooltip yang berisikan data dari kolom description?

-
-
-
-
-
-
catatan:
Sebelumnya saya sudah buat heatmap di ggplot, lalu saya ubah ke ggplotly, namun tooltip yang muncul tidak sesuai dengan axisnya, misal pada axis x 2010, namun tooltip yang muncul dari data axis x 2019. Kemudian karena gagal menemukan cara memperbaiki tooltip tersebut, saya berpindah langsung menggunakan plotly langsung. Hasilnya menggunakan plotly langsung, tooltip default berhasil muncul sesuai data, namun berupa `x:data, y:data, z:data`, nah namun saya belum mendapat cara bagaimana menerapkan isi kolom description pada dataframe ke tooltip plotly tersebut untuk menggantikan tooltip `x,y,z` tersebut.

**Comment by VictorNugraha** [comment, marked as answer]
Halo, @Hertzspr!

Bapak bisa mencoba untuk menambahkan 2 parameter pada fungsi **_plot_ly()_**, kedua parameter itu adalah

- **_text_**: Parameter ini digunakan untuk mengdefine informasi apa saja yang ingin ditampilkan pada tooltip. 
- _**hoverinfo**_: Parameter ini digunakan untuk memaanggil parameter text, parameter ini akan di-isi dengan nama parameter _**text**_.

Berikut contoh codenya:

```
plot_ly(data=df, 
        z=~values, 
        x=~top_year, 
        y=~features, 
        text = ~paste("Deskripsi:", # Pada bagian ini akan di-isi dengan judul informasi yang ingin disampaikan
                      description), # Pada bagian ini akan di-isi dengan nama kolom dari dataframe yang informasinya ingin ditampilkan
        hoverinfo = 'text', # Parameter ini akan di-isi dengan nama parameter text
        type = "heatmap")
```

Demikian informasi yang dapat saya sampaikan, semoga menjawab pertanyaan yang diberkan. Selamat mencoba :D

**Reply by Hertzspr** [reply]
Terima kasih, berhasil tooltipnya.
